packages/oauth/src/index.ts

```diff
@@ -546,6 +579,7 @@ export interface CallbackOptions {
 export interface StateStore {
   // Returned Promise resolves for a string which can be used as an
   // OAuth state param.
+  // TODO: Revisit design. Does installOptions need to be encoded in state if metadata is static?
```

Comment:
If I were you, I would add something like this:

TODO: With bolt-js built-in receivers, `InstallURLOptions` is always static. This means that metadata value is not so useful for customizing OAuth flows per user. If we enhance both this SDK and bolt-js to have a function that accepts request information (e.g., path, headers, query string) plus dynamically generate `InstallURLOptions` per request, developers can easily switch not only medatada but also redirect_uri, scopes, and user_scopes to support multiple installation patterns.